internal/provider/resource_hvn_route.go

```diff
@@ -1,14 +1,293 @@
+				ForceNew:    true,
+			},
+			"hvn_route_id": {
+				Description:      "The ID of the HVN route.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"destination_cidr": {
+				Description:  "The destination CIDR of the HVN route",
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsCIDR,
+			},
+			"target_link": {
+				Description: "A unique URL identifying the target of the HVN route.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			// Computed outputs
+			"self_link": {
+				Description: "A unique URL identifying the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"state": {
+				Description: "The state of the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "The time that the HVN route was created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			// TODO add updated_at after public API returns updated_at for HVN route
+		},
+	}
+}
+
+func resourceHvnRouteCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client := meta.(*clients.Client)
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	destination := d.Get("destination_cidr").(string)
+	hvnRouteID := d.Get("hvn_route_id").(string)
+
+	hvn := d.Get("hvn").(string)
+	var hvnLink *sharedmodels.HashicorpCloudLocationLink
+	hvnLink, err := buildLinkFromURL(hvn, HvnResourceType, loc.OrganizationID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	target := d.Get("target_link").(string)
+	var targetLink *sharedmodels.HashicorpCloudLocationLink
+	// If target type is neither a peering nor a TGW attachment, we will return an error.
+	// Currently create HVN route will only support these two resource types.
+	targetLink, err = buildLinkFromURL(target, PeeringResourceType, loc.OrganizationID)
+	if err != nil {
+		targetLink, err = buildLinkFromURL(target, TgwAttachmentResourceType, loc.OrganizationID)
+		if err != nil {
+			return diag.Errorf("unable to parse target_link for HVN (%s) route with destination CIDR of %s: %v; target must be of either %s or %s", hvnLink.ID, destination, err, PeeringResourceType, TgwAttachmentResourceType)
+		}
+	}
+
+	// Check for an existing HVN.
+	retrievedHvn, err := clients.GetHvnByID(ctx, client, loc, hvnLink.ID)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("unable to find the HVN (%s) for the HVN route", hvnLink.ID)
+		}
+
+		return diag.Errorf("unable to check for presence of an existing HVN (%s): %v", hvnLink.ID, err)
+	}
+
+	if retrievedHvn.State != networkmodels.HashicorpCloudNetwork20200907NetworkStateSTABLE {
```

Comment:
let's please remove this, including checking HVN existence - this is way to much knowledge for the TF provider which is suppose to be a thin client. Especially given that this resource takes an `hvn_link` which is almost guaranty that it won't be just hardcoded but rathe would have an hvn resource/data-source next to it.

Reply:
also, this validation won't happen on `plan` as far as I understand, so there's absolutely no benefits for doing this validation here comparing to calling backend and receiving error from it

Reply:
I agree that checking the HVN state isn't necessary. However, checking for the existence of the resource is an established pattern in all our other resources, as it allows us to provide additional information in the logs, like:
```
2021-05-11T11:15:14.170-0700 [INFO]  plugin.terraform-provider-hcp: 2021/05/11 11:15:14 [INFO] HVN (my-favorite-hvn) not found, proceeding with create: timestamp=2021-05-11T11:15:14.170-0700
2021-05-11T11:15:14.170-0700 [INFO]  plugin.terraform-provider-hcp: 2021/05/11 11:15:14 [INFO] Creating HVN (my-favorite-hvn): timestamp=2021-05-11T11:15:14.170-0700
```

Reply:
discussed with Brenna on zoom and agreed that this is unnecessary here to duplicate backend logic